openapi_tester/schema_tester.py

```diff
@@ -319,19 +380,26 @@ def _test_openapi_type_object(
         case_tester: Optional[Callable[[str], None]],
         ignore_case: Optional[List[str]],
     ) -> None:
-        if "properties" in schema_section:
-            properties = schema_section["properties"]
-        elif "additionalProperties" in schema_section:
+        properties = schema_section.get("properties", {})
+        if "additionalProperties" in schema_section:
             properties = {"": schema_section["additionalProperties"]}
-        else:
-            properties = {}
-        required_keys = schema_section["required"] if "required" in schema_section else list(properties.keys())
-        response_keys = data.keys()
-
-        if len([key for key in response_keys if key in required_keys]) != len(required_keys):
-            missing_keys = ", ".join(str(key) for key in sorted(list(set(required_keys) - set(response_keys))))
-            hint = "Remove the key(s) from your OpenAPI docs, or include it in your API response."
-            message = f"The following properties are missing from the tested data: {missing_keys}."
+
+        required_keys = schema_section.get("required", [])
```

Comment:
Note: The default used to be `list(properties.keys())`, but I've changed it to `{}` so all keys are treated as optional unless specified